app/ide-desktop/lib/client/src/config/parser.ts

```diff
@@ -1,65 +1,80 @@
 // === Help ===
 // ============
 
-let usage =
-    chalk.bold(`\nEnso ${buildCfg.version} command line interface.` + `Usage: enso [options]`) +
+const USAGE =
+    chalk.bold(`\nEnso ${BUILD_INFO.version} command line interface.` + `Usage: enso [options]`) +
     `\n\nBoth single-dash and double-dash prefixes are accepted for all options. For ` +
     `instance, the help message can be displayed by entering either '--help' or '-help'. The ` +
     `'-no-' prefix may be utilized to disable a specific option. For example, to connect to ` +
     `the application from a web-browser, the creation of a window can be suppressed by ` +
     `entering either '-window=false' or '-no-window'.`
 
-class Section {
-    entries: any[] = []
+class Section<T> {
     description = ''
-    constructor(entries: any[] = []) {
-        this.entries = entries
-    }
+    entries: (readonly [cmdOption: string, option: config.Option<T>])[] = []
+}
+
+interface PrintHelpConfig {
+    args: config.Args
+    groupsOrdering: string[]
+    helpExtended: boolean
 }
 
 /** Command line help printer. The `groupsOrdering` parameter specifies the order in which the
  * option groups should be printed. Groups not specified will be printed in the definition order.
  *
- * We use custom help printer because Yargs has several issues:
+ * We use a custom help printer because Yargs has several issues:
  * 1. The option ordering is random and there is no way to enforce it.
  * 2. The option groups ordering is random and there is no way to enforce it.
  * 3. Every option has a `[type`] annotation and there is no API to disable it.
  * 4. There is no option to print commands with single dash instead of double-dash.
  * 5. Help coloring is not supported, and they do not want to support it:
- *    https://github.com/yargs/yargs/issues/251
+ * https://github.com/yargs/yargs/issues/251.
  */
-function printHelp(cfg: { args: config.Args; groupsOrdering: string[]; helpExtended: boolean }) {
-    console.log(usage)
-    const totalWidth = logger.terminalWidth() ?? 80
-    const indentSize = 0
-    const optionPrefix = '-'
-    const spacing = 2
-    const sections: { [key: string]: Section } = {}
+function printHelp(cfg: PrintHelpConfig) {
+    console.log(USAGE)
+    const totalWidth = contentConfig.LOGGER.terminalWidth() ?? DEFAULT_TERMINAL_WIDTH
+    const sections: Record<string, Section<unknown>> = {}
     const topLevelSection = new Section()
     topLevelSection.description =
         'General application switches. For fine-grained control, see the available option groups.'
-    sections['topLevel'] = topLevelSection
+    sections.topLevel = topLevelSection
     for (const groupName of cfg.groupsOrdering) {
         sections[groupName] = new Section()
     }
     let maxOptionLength = 0
 
     for (const [groupName, group] of Object.entries(cfg.args.groups)) {
         let section = sections[groupName]
-        if (section == null) {
+        if (section === undefined) {
```

Comment:
Why this change? I like `== null` more as it tests for both `null` nad `undefined`, which is more secure for possibly future code refactoring. This comment applies to other similar places. If you feel the new form is better, I'd be thankful for short explanation of why :)

Reply:
it should be safe to use `=== undefined` since typescript will error if it's incorrect:
- `null` and `{ any object type } | undefined` have nothing in common
- similar reasoning for accidentally comparing a `T | null` (from `RegExp.match` or `document.getElementById`) against `undefined`

as for why disallow `null`, it's mostly personal preference - plus afaict most if not all of the APIs returning `null`  are legacy APIs.

i have nothing against `== null` though, and i'd be happy to switch back. the reason why i chose to do things this way is because of the `eqeqeq` rule that prefers `===` over `==` - but it has an option to prefer `== null` so it'd be *very* simple to change back.

Reply:
`plus afaict most if not all of the APIs returning null are legacy APIs.` - oh, this is something new to me. If you want to return lack of value, what should you return? I'm never returning `undefined` in my APIs, as I learned that `undefined` means lack of definition in JS world, and it should never be used as a real value. 

After your explanation I'm 100% OK with this version, thanks for it. Le]ts keep it this way, sounds more "rigid" than my proposal :)